controllers/tasks.js

```diff
@@ -205,6 +205,36 @@ const fetchTasks = async (req, res) => {
   }
 };
 
+/**
+ * Fetches all the tasks of the logged in user
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getTasksByUser = async (req, res) => {
+  try {
+    const { username, id: authenticatedUserId } = req.userData;
+    const requestedUserId = req.params.userId;
+    const dev = req.query.dev;
+
+    if (requestedUserId === authenticatedUserId && dev) {
+      if (req.query.completed) {
+        const allCompletedTasks = await tasks.fetchUserCompletedTasks(username);
+        return res.json(allCompletedTasks);
+      } else {
+        const allTasks = await tasks.fetchUserTasks(username, []);
+        return res.json(allTasks);
+      }
+    } else {
+      return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
```

Comment:
Shouldn't this be forbidden instead of internal server error?
Status 403 instead of 500, and also have error message like `access denied` since the authenticated user is attempting to access another user's task data through this route.

Reply:
The current implementation assumes that the only valid scenario for this route is when the requestedUserId matches the authenticatedUserId with the dev query parameter. Any other scenario is treated as an unexpected state, and thus, a 500 Internal Server Error is returned.
The intention here is that this condition indicates a logical flaw in the request handling, not merely a user authorization issue.

Reply:
I think we should handle these errors gracefully instead of returning an internal server error, which suggests a backend issue prevented processing the request. We should specify the exact reason for failure, like unauthorised access. This would make debugging easier later if someone made a logical flaw and hit this error.

Reply:
added 403